File: app/src/main/java/com/cheatdatabase/cheat_detail_view/CheatViewFragmentAdapter.java
package com.cheatdatabase.cheat_detail_view;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.cheatdatabase.data.model.Cheat;
import com.cheatdatabase.data.model.Game;

import java.util.List;

public class CheatViewFragmentAdapter extends FragmentPagerAdapter {

    private List<Cheat> cheatArray;
    private Game game;

    public CheatViewFragmentAdapter(FragmentManager fm, Game game, List<Cheat> cheatArray) {
        super(fm);
        this.game = game;
        this.cheatArray = cheatArray;
    }

    @Override
    public Fragment getItem(int position) {
        return CheatViewFragment.newInstance(game, position);
    }

    @Override
    public int getCount() {
        return cheatArray.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return cheatArray.get(position).getCheatTitle();
    }

}

File: app/src/main/java/com/cheatdatabase/cheatdetailview/CheatViewFragmentAdapter.kt
package com.cheatdatabase.cheatdetailview

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.cheatdatabase.cheatdetailview.CheatViewFragment.Companion.newInstance
import com.cheatdatabase.data.model.Cheat
import com.cheatdatabase.data.model.Game

class CheatViewFragmentAdapter(
    fragmentActivity: FragmentActivity,
    private val game: Game,
    private val cheatArray: ArrayList<Cheat>
) : FragmentStateAdapter(fragmentActivity) {
    override fun createFragment(position: Int): Fragment {
        return newInstance(game, position)
    }

    override fun getItemCount(): Int {
        return cheatArray.size
    }

//    override fun getPageTitle(position: Int): CharSequence? {
//        return cheatArray[position].cheatTitle
//    }
}